packages/amplify_authenticator/lib/src/enums/authenticator_step.dart

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/// The current step of the authentication flow.
+enum AuthenticatorStep {
+  /// The authenticator is in a loading state.
+  ///
+  /// Either Amplify.configure() or the current authentication status
+  /// is pending.
+  loading,
+
+  /// The user is authenticated.
+  authenticated,
+
+  /// The user is on the Landing Page step.
+  ///
+  /// This step can be used to create and initial landing page
+  /// that will be displayed to users prior to showing sign in / sign up.
+  ///
+  /// This step is only relevant for custom authenticator UIs.
+  /// The prebuilt authenticator uses the sign in view as the landing page.
+  landingPage,
```

Comment:
Personally I think the landingPage is a good feature. @dnys1 Can you elaborate a bit on what you mean by 'handled by loading'?

Side note (doesn't need to be addressed now): I think it might be helpful to display the loading widget while API calls are completing. These are very fast right now in the default flow under good network conditions, but with slow network connections or when we introduce custom flows with lambda triggers these calls might take a bit longer.

Reply:
I worded that poorly. Maybe I don't fully understand the use case of it, but it seems like a half-baked feature. If it's to show a splash screen, that should be done for `loading` and if it's for onboarding then that should follow authentication anyway.

There are loading indicators already, though? Flashing a full loading screen for a couple seconds continuously seems like very poor UX.

Reply:
The docs I added might be describing the intended use case poorly. I added an example [here](https://github.com/aws-amplify/amplify-flutter/pull/1281/files#r787951259). If that clears it up, I can improve the doc comments here.